src/platformio.ts

```diff
@@ -85,21 +98,124 @@ export class PlatformIO {
+            if (autoUpdateIncludes && !this._autoUpdateIncludesEnabled) {
+                this.buildFolderList();
+                const configFilePath = path.join(vscode.workspace.rootPath, ".vscode", "c_cpp_properties.json");
+                if (!Utility.fileExistsSync(configFilePath)) {
+                    this.addIncludePath(true);
+                }
+
+                // Watch .piolibdeps and lib folders
+                const fileSystemWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**/{.piolibdeps,lib}/**");
+                this._autoUpdateIncludesDisposables.push(fileSystemWatcher);
+                fileSystemWatcher.onDidCreate((e: vscode.Uri) => this.fileSystemDidCreate(e), this, this._autoUpdateIncludesDisposables);
+                fileSystemWatcher.onDidDelete((e: vscode.Uri) => this.fileSystemDidDelete(e), this, this._autoUpdateIncludesDisposables);
+
+                // Watch platformio.ini file
+                const iniFileWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**/platformio.ini");
+                this._autoUpdateIncludesDisposables.push(iniFileWatcher);
+                iniFileWatcher.onDidChange((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+                iniFileWatcher.onDidCreate((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+            }
+
+            if (!autoUpdateIncludes && this._autoUpdateIncludesEnabled) {
+                this._folderList = undefined;
+                this.disposeAutoUpdateIncludes();
+            }
+
+            this._autoUpdateIncludesEnabled = autoUpdateIncludes;
+        }
+    }
+
+    private fileSystemDidCreate(e: vscode.Uri): void {
+        let parent = this.getParentFolder(e.path);
+        if (!this._folderList.has(parent)) {
+            this._folderList.add(parent);
+            this.debounceChanges();
+        }
+
+        try {
+            if (fs.statSync(e.fsPath).isDirectory()) {
+                if (!this._folderList.has(e.path)) {
+                    this._folderList.add(e.path);
+                    this.debounceChanges();
+                }
+            }
+        } catch (error) {
+            // Ignore errors, the file or folder was probably temporary and was
+            // renamed or deleted before we could check if it was a folder.
+        }
+    }
+
+    private fileSystemDidDelete(e: vscode.Uri): void {
+        if (this._folderList.has(e.path)) {
+            this._folderList.delete(e.path);
+            this._folderList.forEach((value: string) => {
+                if (value.startsWith(e.path)) {
+                    this._folderList.delete(value);
+                }
+            });
+            this.debounceChanges();
+        }
+    }
+
+    private getParentFolder(path: string): string {
+        let reg = path.match(/^(.+)\/([^/]+)$/);

```

Comment:
That is why I use the ```vscode.Uri.path``` which will always have forward slashes on all platforms. You can see I use ```vscode.Uri.fsPath``` when calling Node fs functions, because that has the platform specific path.